pkg/ccl/backupccl/restore_job.go

```diff
@@ -1189,10 +1189,18 @@ func createImportingDescriptors(
 			}
 
 			if len(details.Tenants) > 0 {
-				initialTenantZoneConfig, err := sql.GetHydratedZoneConfigForTenantsRange(ctx, txn)
+				// Creating a new collection with system codec the purpose of resolving
+				// zone config for tenants range. It won't query any table descriptor
+				// from collection since it's only hydrating zone configs for tenant
+				// ranges which doesn't have corresponding descriptor. But, it's still
+				// to release everything after.
+				cf := descs.NewBareBonesCollectionFactory(r.settings, keys.SystemSQLCodec)
+				descriptors := cf.NewCollection(ctx)
```

Comment:
We already have a collection here in the form of `descsCol`. In any case you almost never want to call `NewBareBonesCollectionFactory`, instead if a collection is not available in a txn you should rework how the txn is opened by using `sql.DescsTxn`.

Reply:
heh, I totally missed `descCol` >_> because my instinct was to have goland jump to the function signature :(, thanks for catching that.

Reply:
Fixed. Thanks again.